### ISSUE TITLE
[apex] ApexCRUDViolation not checking queries when used as map constructor arguments

### ISSUE THREAD

**Issue by jonathanwiesel** [pmd/pmd#3419]
**Affects PMD Version:** 
5.5.3 - 6.36.0

**Rule:**

[ApexCRUDViolation](https://pmd.github.io/latest/pmd_rules_apex_security.html#apexcrudviolation)

**Description:**

When a query is issued inside a map constructor the CRUD check is not enforced

**Code Sample demonstrating the issue:**

```
public class Foo {
  public void bar() {
    Map<Id, Account> accs = new Map<Id, Account>([SELECT Name FROM Account]);
  }
}

```

**Expected outcome:**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *Apex PMD VS Code Plugin*

**Comment by jonathanwiesel** [comment]
Related to #3378 due to the difficulty of determining the appropriate type to check on maps
